<launch>
  <arg name="input_autoware_engage" default="/autoware/engage" />
  <arg name="input_vehicle_control_mode" default="/vehicle/status/control_mode" />
  <arg name="input_is_emergency" default="/system/emergency/is_emergency" />
  <arg name="input_route" default="/planning/mission_planning/route" />
  <arg name="input_twist" default="/localization/twist" />

  <arg name="output_autoware_state" default="/autoware/state" />
  <arg name="output_autoware_engage" default="/autoware/engage" />

  <arg name="config_file" default="$(find autoware_state_monitor)/config/autoware_state_monitor.yaml" />

  <node pkg="autoware_state_monitor" type="autoware_state_monitor" name="autoware_state_monitor" output="screen">
    <remap from="~input/autoware_engage" to="$(arg input_autoware_engage)"/>
    <remap from="~input/vehicle_control_mode" to="$(arg input_vehicle_control_mode)"/>
    <remap from="~input/is_emergency" to="$(arg input_is_emergency)"/>
    <remap from="~input/route" to="$(arg input_route)"/>
    <remap from="~input/twist" to="$(arg input_twist)"/>

    <remap from="~output/autoware_state" to="$(arg output_autoware_state)"/>
    <remap from="~output/autoware_engage" to="$(arg output_autoware_engage)"/>

    <rosparam command="load" file="$(arg config_file)" />

    <param name="update_rate" value="10.0" />
    <param name="th_arrived_distance_m" value="1.0" />
    <param name="th_stopped_time_sec" value="1.0" />
    <param name="th_stopped_velocity_mps" value="0.01" />

    <param name="disengage_on_route" value="true" />
    <param name="disengage_on_complete" value="false" />
    <param name="disengage_on_emergency" value="false" />
  </node>
</launch>
